### PR TITLE
Fix to overwrite excel files having a single sheet

### PR DESCRIPTION
**Pull request by aparajit-pratap**
### Purpose

This fixes the issue: http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-7515 with overwriting excel files containing only a single worksheet. If the `Excel.WriteToFile` node is executed with the `overwrite` argument set to `true`, it needs to clear the existing sheet and replace it with a new one of the same name with the new data being input to the node. However if there is only a single worksheet in the file, it throws an error. 

The fix is to add a new worksheet, then delete the existing one, followed by renaming the new worksheet to the one that was deleted.

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] The level of testing this PR includes is appropriate

### Reviewers

@sharadkjaiswal  Reviewer 1  

### FYIs

@monikaprabhu

### CODE REVIEW

File: test/Libraries/DynamoMSOfficeTests/ExcelTests.cs [hunk 1]
```diff
@@ -984,6 +984,31 @@ public void TestWriteFunctionObjectToExcel()
...
+
+            var writeNode = ViewModel.Model.CurrentWorkspace.GetDSFunctionNodeFromWorkspace("Excel.WriteToFile");
+
+            Assert.IsTrue(File.Exists(filename.Value));
+
+            Assert.IsTrue(writeNode.CachedValue.IsCollection);
+            var list = writeNode.CachedValue.GetElements();
+
+            Assert.AreEqual(3, list.Count);
+            AssertPreviewValue(writeNode.GUID.ToString(), new object[] { new object[] { 1 }, new object[] { 2 }, new object[] { 3 } });
```
Comment (sharadkjaiswal) [inline test/Libraries/DynamoMSOfficeTests/ExcelTests.cs]: You might also want to verify that the xls file has only one sheet with name 'Sheet1'

Comment (monikaprabhu) [inline test/Libraries/DynamoMSOfficeTests/ExcelTests.cs]: I think that should be seperate test, if there is a single sheet what happens - I'll add the test as part of verify the bug I reported -http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-7515

Comment (aparajit-pratap) [inline test/Libraries/DynamoMSOfficeTests/ExcelTests.cs]: The excel has only one sheet. I know coz I created it. There's no need to verify this.

File: test/Libraries/DynamoMSOfficeTests/ExcelTests.cs [hunk 2]
```diff
@@ -984,6 +984,33 @@ public void TestWriteFunctionObjectToExcel()
...
+
+            var wb = Excel.ReadExcelFile(filename.Value);
+            Assert.IsTrue(wb.WorkSheets.Length == 1);
```
Comment (aparajit-pratap) [inline test/Libraries/DynamoMSOfficeTests/ExcelTests.cs]: Done

**Pull request merged**